contracts/farm/MasterBarista.sol

```diff
@@ -669,7 +687,7 @@ contract MasterBarista is IMasterBarista, OwnableUpgradeable, ReentrancyGuardUpg
 
     // Burn BEAN if user emergencyWithdraw LATTE
     if (_stakeToken == address(latte)) {
-      bean.burn(_msgSender(), user.amount);
+      bean.burn(_for, user.amount);
```

Comment:
This should change in `depositLatte()` function to mint bean to `for`to mint it to `_msgSender` then leave this one as is.